numpyro/infer/util.py

```diff
@@ -357,6 +357,21 @@ def _guess_max_plate_nesting(model_trace):
     return max_plate_nesting
 
 
+# TODO: follow pyro.util.check_site_shape logics for more complete validation
```

Comment:
Do distribution dims follow the same convention as in Pyro, i.e. [enum dims][batch dims][event dims], where the number of `batch_dims` equals the maximum number of nested plates in the model? I would like to better understand what class of errors would this validation logic catch and what would it miss.

Reply:
Yes, it follows the same convention (except for `scan`, we have `[time_dim][enum dims][batch dims][event dims]`). (this validation does not check for `enum dims` - which is defered to funsor logic)

Pyro validation has some logic to check if some expected shape is met. This only checks the number of batch dimensions agrees with the dimension of the leftmost plate.